src/fwrite.c

```diff
@@ -1067,36 +1073,29 @@ void fwriteMain(fwriteMainArgs args)
         }
       }
     }
-    if (args.is_gzip) {
-#ifndef NOZLIB
-      deflateEnd(mystream);
-#endif
-    }
 
   } // end of parallel for loop
 
+  free(buffPool);
+
+#ifndef NOZLIB
+  free(zbuffPool);
+
 /* put a 4-byte integer into a byte array in LSB order */
 #define PUT4(a,b) ((a)[0]=(b), (a)[1]=(b)>>8, (a)[2]=(b)>>16, (a)[3]=(b)>>24)
 
   // write gzip tailer with crc and len
-    if (args.is_gzip) {
-#ifndef NOZLIB
-      unsigned char tail[10];
-      tail[0] = 3;
-      tail[1] = 0;
-      PUT4(tail + 2, crc);
-      PUT4(tail + 6, len);
-      int ret = WRITE(f, tail, 10);
-      compress_len += 10;
-      if (ret == -1)
-        STOP("Error: can't write gzip tailer"); // # nocov
-#endif
-    }
-
-  free(buffPool);
-#ifndef NOZLIB
-  free(thread_streams);
-  free(zbuffPool);
+  if (args.is_gzip) {
+    unsigned char tail[10];
+    tail[0] = 3;
+    tail[1] = 0;
+    PUT4(tail + 2, crc);
+    PUT4(tail + 6, len);
+    int ret = WRITE(f, tail, 10);
+    compress_len += 10;
+    if (ret == -1)
+      STOP("Error: can't write gzip tailer"); // # nocov
```

Comment:
```suggestion
      STOP("Error: can't write gzip trailer"); // # nocov
```
Does this message need translation like other calls to `STOP(...)`?

Reply:
I thought 'tailer' was a special word 😂 

Yes, added `_()`